src/main/java/ru/yandex/practicum/filmorate/storage/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,183 @@
+                user.getBirthday(),
+                user.getId());
+        final String checkUserQuery = "SELECT * FROM users WHERE user_id = ?";
+        SqlRowSet userRows = jdbcTemplate.queryForRowSet(checkUserQuery, user.getId());
+        if (!userRows.next()) {
+            log.warn("Пользователь с идентификатором {} не найден.", user.getId());
+            throw new DataNotFoundException("Пользователь с идентификатором " + user.getId() + " не найден.");
+        }
+        log.debug("Обновлён пользователь c ID: " + user.getId());
+        return user;
+    }
+
+    @Transactional
+    @Override
+    public User getUserById(int id) {
+        log.debug("getUserById({}).", id);
+        String sqlQuery = "SELECT * FROM users WHERE user_id = ?;";
+        SqlRowSet rs = jdbcTemplate.queryForRowSet(sqlQuery, id);
+        if (rs.next()) {
+            return User.builder()
+                    .id(rs.getInt(USER_ID))
+                    .email(rs.getString(EMAIL))
+                    .login(rs.getString(LOGIN))
+                    .name(rs.getString(USER_NAME))
+                    .birthday(Objects.requireNonNull(rs.getDate(BIRTHDAY)).toLocalDate())
+                    .build();
+        } else {
+            throw new DataNotFoundException("Пользователь с id = " + id + " не найден.");
+        }
+    }
+
+    @Transactional
+    @Override
+    public void deleteUserById(int id) {
+        log.debug("deleteUser({}).", id);
+        String sqlQuery = "DELETE FROM users WHERE user_id = ?;";
+        if (id > 0) {
+            jdbcTemplate.update(sqlQuery, id);
+            log.debug("Удален пользователь с ID: {}", id);
+        } else
+            throw new DataNotFoundException("Пользователь с ID=" + id + " не найден!");
+    }
+
+    @Transactional
+    @Override
+    public List<User> getUsersFriendsById(int id) {
+        log.debug("getUsersFriends({}).", id);
+        String sqlQuery = "select u.user_id, u.email, u.name, u.login, u.birthday " +
+                "from friends as f left join users as u " +
+                "on f.other_user_id = u.user_id where f.user_id = ?" +
+                "order by u.user_id";
+
+        List<User> friends = jdbcTemplate.query(sqlQuery, (rs, rowNum) -> makeUser(rs), id);
+        log.info("Получен пользователь c ID={}", id);
+        return friends;
+    }
+
+    @Transactional
+    @Override
+    public void deleteFriend(int id, int friendId) {
+        log.debug("deleteFriend({}, {}).", id, friendId);
+        String sqlQuery = "DELETE FROM friends WHERE user_id = ? AND other_user_id = ? ;";
+        jdbcTemplate.update(sqlQuery, id, friendId);
+        log.debug("Пользователь с ID: {} удалил из друзей Пользователь с ID: {}", id, friendId);
+    }
+
+    @Transactional
+    @Override
+    public List<User> getMutualFriends(int userId, int otherUserId) {
+        log.debug("getMutualFriends({}, {}).", userId, otherUserId);
+        String query = "SELECT u.* FROM friends f "
```

Comment:
"select * from USERS, FRIENDS where USERS.USER_ID = FRIENDS.FRIEND_ID AND FRIENDS.USER_ID = ?"